src/sagemaker/model.py

```diff
@@ -546,6 +547,10 @@ def register(
             if self.model_data is not None:
                 container_def["ModelDataUrl"] = self.model_data
 
+        if self.model_type is JumpStartModelType.PROPRIETARY:
+            source_uri = self.model_package_arn
+            model_package_group_name = self.model_id
```

Comment:
Instead of having this logic here, we should have logic in jumpstart itself to set model group name. 

Also where are we setting sourceUri from jumpstart

Reply:
here on line 551 model_package_arn contains the model package arn of the jumpstart proprietary model,